v4.0.x: `--enable-install-libpmix` in a tarball does not work

While doing some testing for #6227, I noticed that `./configure --enable-install-libpmix` does not work from a tarball created from both master and the v4.0.x branch (I did not test farther back than this).  

Specifically, when you `make`, you get:

```
...
make[4]: Entering directory `/home/jsquyres/git/ompi/openmpi-gitclone/opal/mca/pmix/pmix4x/pmix/man'
../contrib/md2nroff.pl --source=pmix_init.3.md;
../contrib/md2nroff.pl --source=pmix_finalize.3.md;
../contrib/md2nroff.pl --source=pmix_initialized.3.md;
../contrib/md2nroff.pl --source=pmix_abort.3.md;
../contrib/md2nroff.pl --source=pmix_put.3.md;
../contrib/md2nroff.pl --source=pmix_commit.3.md;
../contrib/md2nroff.pl --source=pmix.7.md;
../contrib/md2nroff.pl --source=pmix_constants.7.md;
Source file does not exist (pmix_init.3.md) at ../contrib/md2nroff.pl line 38.
Source file does not exist (pmix_initialized.3.md) at ../contrib/md2nroff.pl line 38.
Source file does not exist (pmix_finalize.3.md) at ../contrib/md2nroff.pl line 38.
make[4]: *** [man3/pmix_init.3] Error 2
...
```

I note that this option works when you use it in a git clone (i.e., not a tarball build).  So it's probably something missing from `make dist`.

@rhc54 Is it expected that this CLI option is supposed to work?  If so, we should fix it. Otherwise, we should probably remove this CLI option.

FYI @georgemarselis

New testing:

* With openmpi-master-201907130242-020a591: `--enable-install-libpmix` works fine.
* With openmpi-v4.0.x-201907100241-d3a7360: `--enable-install-libpimx` fails in configure with the following:

```
shell$ ./configure --prefix=$bogus --enable-install-libpmix
...lots of output...
--- MCA component pmix:pmix3x (m4 configuration macro)
checking for MCA component pmix:pmix3x compile mode... dso
checking if PMIx timing is enabled... no (disabled)
checking if want to install standalone libpmix... yes
configure: OPAL configuring in opal/mca/pmix/pmix3x/pmix
configure: running /bin/sh './configure' --disable-debug --disable-pmix-timing --without-tests-examples --disable-pmix-binaries --disable-pmix-backward-compatibility --disable-visibility --enable-embedded-libevent --with-libevent-header=\"opal/mca/event/libevent2022/libevent2022.h\" --enable-embedded-hwloc --with-hwloc-header=\"opal/mca/hwloc/hwloc201/hwloc201.h\"  '--prefix=/home/jsquyres/bogus' '--enable-install-libpmix' 'CFLAGS=-O3 -DNDEBUG  ' 'CPPFLAGS=-I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360 -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360 -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/include -I/home/jqsuyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/include -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/event/libevent2022/libevent -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/event/libevent2022/libevent/include -I/home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/hwloc/hwloc201/hwloc/include  ' --cache-file=/dev/null --srcdir=. --disable-option-checking
configure: builddir: /home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/pmix/pmix3x/pmix
configure: srcdir: /home/jsquyres/tmp/openmpi-v4.0.x-201907100241-d3a7360/opal/mca/pmix/pmix3x/pmix

============================================================================
== Configuring PMIx
============================================================================
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu

...lots more output...

============================================================================
== Libevent
============================================================================
checking for libevent in... (default search paths)
looking for header in /usr/include
checking event.h usability... no
checking event.h presence... no
checking for event.h... no
looking for header in /usr/include/include
checking event.h usability... no
checking event.h presence... no
checking for event.h... no
configure: WARNING: LIBEVENT SUPPORT NOT FOUND
configure: error: CANNOT CONTINUE
configure: /bin/sh './configure' *failed* for opal/mca/pmix/pmix3x/pmix
checking if v3.x component is to be used... yes - using the internal v3.x library
configure: WARNING: INTERNAL PMIX FAILED TO CONFIGURE
configure: error: CANNOT CONTINUE
```

That is: it fails in the libevent configury inside the pmix configury inside the pmix:pmix3x configury.

--- a/opal/mca/pmix/pmix3x/configure.m4
+++ b/opal/mca/pmix/pmix3x/configure.m4
@@ -12,7 +12,7 @@
 #                         All rights reserved.
 # Copyright (c) 2011-2013 Los Alamos National Security, LLC.
 #                         All rights reserved.
-# Copyright (c) 2010-2017 Cisco Systems, Inc.  All rights reserved.
+# Copyright (c) 2010-2020 Cisco Systems, Inc.  All rights reserved
 # Copyright (c) 2013-2019 Intel, Inc.  All rights reserved.
 # Copyright (c) 2015-2016 Research Organization for Information Science
 #                         and Technology (RIST). All rights reserved.
@@ -37,10 +37,6 @@ AC_DEFUN([MCA_opal_pmix_pmix3x_CONFIG],[
     opal_pmix_pmix3x_save_LDFLAGS=$LDFLAGS
     opal_pmix_pmix3x_save_LIBS=$LIBS
 
-    AC_ARG_ENABLE([install-libpmix],
-                  [AC_HELP_STRING([--enable-install-libpmix],
-                                  [Enable a native PMIx library and headers in the OMPI install location (default: disabled)])])
-
     AC_ARG_ENABLE([pmix-timing],
                   [AC_HELP_STRING([--enable-pmix-timing],
                                   [Enable PMIx timing measurements (default: disabled)])])
@@ -54,16 +50,12 @@ AC_DEFUN([MCA_opal_pmix_pmix3x_CONFIG],[
     fi
 
     opal_pmix_pmix3x_args="$opal_pmix_pmix3x_timing_flag --without-tests-examples --disable-pmix-binaries --disable-pmix-backward-compatibility --disable-visibility --enable-embedded-libevent --with-libevent-header=\\\"opal/mca/event/$opal_event_base_include\\\" --enable-embedded-hwloc --with-hwloc-header=\\\"$opal_hwloc_base_include\\\""
+    opal_pmix_pmix3x_args="--with-pmix-symbol-rename=OPAL_MCA_PMIX3X_ --enable-embedded-mode $opal_pmix_pmix3x_args"
     AS_IF([test "$enable_debug" = "yes"],
           [opal_pmix_pmix3x_args="--enable-debug $opal_pmix_pmix3x_args"
            CFLAGS="$OPAL_CFLAGS_BEFORE_PICKY $OPAL_VISIBILITY_CFLAGS -g"],
           [opal_pmix_pmix3x_args="--disable-debug $opal_pmix_pmix3x_args"
            CFLAGS="$OPAL_CFLAGS_BEFORE_PICKY $OPAL_VISIBILITY_CFLAGS"])
-    AC_MSG_CHECKING([if want to install standalone libpmix])
-    AS_IF([test "$enable_install_libpmix" = "yes"],
-          [AC_MSG_RESULT([yes])],
-          [AC_MSG_RESULT([no])
-           opal_pmix_pmix3x_args="--with-pmix-symbol-rename=OPAL_MCA_PMIX3X_ --enable-embedded-mode $opal_pmix_pmix3x_args"])
     AS_IF([test "$with_devel_headers" = "yes"],
           [opal_pmix_pmix3x_args="--with-devel-headers  $opal_pmix_pmix3x_args"])
     CPPFLAGS="-I$OPAL_TOP_SRCDIR -I$OPAL_TOP_BUILDDIR -I$OPAL_TOP_SRCDIR/opal/include -I$OPAL_TOP_BUILDDIR/opal/include $CPPFLAGS"

